tools/newrelic-infra-public-keys/newrelic-infra-public-keys

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+KEYRING_FILE_NAME=newrelic-infra-keyring.gpg
+TRUSTED_GPG_D_KEYRING=/etc/apt/trusted.gpg.d/${KEYRING_FILE_NAME}
+USR_SHARE_KEYRING=/usr/share/keyrings/${KEYRING_FILE_NAME}
+KEYS_LOCATION=/opt/newrelic-infra-public-keys
+
+if [ ! -f ${USR_SHARE_KEYRING} ]; then
+    touch ${USR_SHARE_KEYRING}
+fi
+chmod a+r ${USR_SHARE_KEYRING}
+
+cat "${KEYS_LOCATION}/keys/current/newrelic_apt_key_current.gpg" | gpg -q --no-default-keyring --keyring ${USR_SHARE_KEYRING} --import --batch
+cat "${KEYS_LOCATION}/keys/next/newrelic_apt_key_next.gpg" | gpg -q --no-default-keyring --keyring ${USR_SHARE_KEYRING} --import --batch
+
+# backward compatibility hack
+if  ! grep "signed-by=${USR_SHARE_KEYRING}" /etc/apt/sources.list.d/newrelic-infra.list 2>/dev/null 1>/dev/null; then
```

Comment:
What if I have a commented out line with `signed-by` on that file?

Reply:
I'm not sure we should be relying on a naive check for something of this importance. If users have, for example, a commented line on this file, it could trip the if and install the keyring in the wrong place.

On top of that, we don't have guarantees of `newrelic-infra.list` being the, or the only place where our repo is defined. Users may move or rename things while troubleshooting.

While those may be corner cases, I think they are important because this script is not visible or editable for end users. It could simply fail and there wouldn't be any actionable error.